prefect_bitbucket/repository.py

```diff
@@ -0,0 +1,190 @@
+"""Allows for interaction with a BitBucket repository.
+
+The `BitBucket` class in this collection is a storage block that lets Prefect agents
+pull Prefect flow code from BitBucket repositories.
+
+The `BitBucket` block is ideally configured via the Prefect UI, but can also be used
+in Python as the following examples demonstrate.
+
+Examples
+--------
+```python
+    from prefect_bitbucket.repository import BitBucketRepository
+
+    # public BitBucket repository
+    public_bitbucket_block = BitBucketRepository(
+        repository="https://bitbucket.com/my-project/my-repository.git"
+    )
+
+    public_bitbucket_block.save(name="my-bitbucket-block")
+
+
+    # specific branch or tag
+    branch_bitbucket_block = BitBucketRepository(
+        reference="branch-or-tag-name",
+        repository="https://bitbucket.com/my-project/my-repository.git"
+    )
+
+    branch_bitbucket_block.save(name="my-bitbucket-block")
+
+
+    # private BitBucket repository
+    private_bitbucket_block = BitBucketRepository(
+        repository="https://bitbucket.com/my-project/my-repository.git",
+        bitbucket_credentials=BitBucketCredentials.load("my-bitbucket-credentials-block")
+
+
+    )
+
+    private_bitbucket_block.save(name="my-private-bitbucket-block")
+```
```

Comment:
I think this needs unindent.
![image](https://user-images.githubusercontent.com/15331990/204884586-b53019f3-7a3d-45a4-8c77-12c79e805e25.png)

```suggestion

Examples
```python
from prefect_bitbucket.repository import BitBucketRepository

# public BitBucket repository
public_bitbucket_block = BitBucketRepository(
    repository="https://bitbucket.com/my-project/my-repository.git"
)

public_bitbucket_block.save(name="my-bitbucket-block")


# specific branch or tag
branch_bitbucket_block = BitBucketRepository(
    reference="branch-or-tag-name",
    repository="https://bitbucket.com/my-project/my-repository.git"
)

branch_bitbucket_block.save(name="my-bitbucket-block")


# private BitBucket repository
private_bitbucket_block = BitBucketRepository(
    repository="https://bitbucket.com/my-project/my-repository.git",
    bitbucket_credentials=BitBucketCredentials.load("my-bitbucket-credentials-block")


)

private_bitbucket_block.save(name="my-private-bitbucket-block")
```